### PR TITLE
[xh] Add Oracle dockerfile to support thick mode

### PR DESCRIPTION
**Pull request by matrixstone**
# Description
- Create dockerfile for oracle image to install the Oracle instant client, which is required to enable Oracle thick mode.
- Add 'mode' config in both Oracle db destination and source to specify the connection mode as thick or thin.

# How Has This Been Tested?
Tested with Mage UI to see instant client enabled.
Screenshot:
WX20240122-005042@2x

# Checklist
- [X] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [X] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] If new documentation has been added, relative paths have been added to the appropriate section of `docs/mint.json`

cc:
@wangxiaoyou1993

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -106,3 +108,7 @@ services:
       dockerfile: ./dev.spark.Dockerfile
     depends_on:
       - server_spark
+networks:
```
Comment (wangxiaoyou1993) [inline docker-compose.yml:111]: You don't need to modify the docker-compose.yaml

File: docs/design/data-loading.mdx [hunk 2]
```diff
@@ -816,6 +817,7 @@ Initializes the OracleDB data loading client.
     -   `host (str)`: Host address for database.
     -   `port (int)`: Port on which the database is running Defaults to 3306.
     -   `service (str)`: OracleDB service
+    -   `mode (str)`: switch between OracleDB mode of thin or thick
```
Comment (wangxiaoyou1993) [inline docs/design/data-loading.mdx:820]: Can you wrap the possible values with backticks: Switch between OracleDB mode of `thin` or `thick`

File: mage_integrations/mage_integrations/destinations/oracledb/__init__.py [hunk 3]
```diff
@@ -45,10 +45,14 @@ def password(self) -> str:
     def database(self) -> str:
         return self.config['database']
 
+    @property
+    def mode(self) -> str:
+        return self.config['mode'] or 'thin'
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/destinations/oracledb/__init__.py:50]: `return self.config.get('mode') or 'thin'

File: mage_integrations/mage_integrations/destinations/oracledb/templates/config.json [hunk 4]
```diff
@@ -4,6 +4,6 @@
     "service": "",
     "password": "",
     "user": "",
-    "database": ""
-}
-
+    "database": "",
+    "mode": "thin"
+}
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/destinations/oracledb/templates/config.json:9]: nit: add empty line at the end of flle

File: mage_integrations/mage_integrations/sources/oracledb/__init__.py [hunk 5]
```diff
@@ -26,6 +26,10 @@ def user(self) -> str:
     def password(self) -> str:
         return self.config['password']
 
+    @property
+    def mode(self) -> str:
+        return self.config['mode'] or 'thin'
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/oracledb/__init__.py:31]: same here. `self.config.get('mode') or 'thin'`

**Review by wangxiaoyou1993 — COMMENTED** [review]
Can you also update the doc to mention the custom Dockerfile needs to be used to use "thick" mode?
Also, can you also update io.oracle class for the thick mode change?

File: docker-compose.yml (hunk 1 above)
Comment (matrixstone) [reply docker-compose.yml:111]: Removed

**Review by wangxiaoyou1993 — APPROVED** [review]

**Comment by faz-ic** [comment]
Think its worth noting that the reason this didn't work when I was testing / building out was due to CPU architecture (running on M1 ARM64 vs AMD x64). This is currently overcome by having two separate docker files when developing locally vs when deployed. 

Been meaning to mention this for some time. Food for thought - @wangxiaoyou1993

**Pull request merged**